File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.js
const checkItem = (arr) => {
    const stack = [];
    
    for (let i = 0; i < arr.length; i++) {
        if (!stack.length) stack.push(arr[i]);
        else {
            if (stack[stack.length - 1] === '[' && arr[i] === ']') {
                stack.pop();
            } else if (stack[stack.length - 1] === '(' && arr[i] === ')') {
                stack.pop();
            } else if (stack[stack.length - 1] === '{' && arr[i] === '}') {
                stack.pop();
            } else {
                stack.push(arr[i])
            }
        }
    }
    
    return stack.length ? false : true;
}

function solution(s) {
    s = s.split("");
    let rotationCount = 0;
    let ans = 0;
    
    while (rotationCount !== s.length) {
        const result = checkItem(s);
        if (result) ans++;
        
        s.push(s.shift());
        rotationCount++;
    }
    
    return ans;
}

File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
from collections import deque

def solution(s):
    answer = 0
    s = deque(s)
    
    for i in range(len(s)):
        s.append(s.popleft())
        
        stack = []
        
        for item in s:
            if not len(stack):
                stack.append(item)
            else:
                if item == "]":
                    if stack[-1] == "[":
                        stack.pop()
                elif item == ")":
                    if stack[-1] == "(":
                        stack.pop()
                elif item == "}":
                    if stack[-1] == "{":
                        stack.pop()
                else:
                    stack.append(item)
            
        if not len(stack):
            answer += 1


    return answer